nvtabular/workflow/workflow.py

```diff
@@ -112,6 +109,9 @@ def transform(self, dataset: Dataset) -> Dataset:
         -------
         Dataset
         """
+        if not self.output_schema:
+            self.fit(dataset)
```

Comment:
Running the full fit here doesn't seem quite right to me - I'd rather either throw an exception (saying that you should run fit first) or call fit_schema instead of fit.

Reply:
Hmm, okay. What's the downside of running the full `fit` here? It seems like it provides a convenience from the standpoint of the calling code, but maybe I'm missing a negative impact of some kind.

Reply:
(Made the change, but still curious.)

Reply:
I'd find it surprising if calling transform (without calling fit) automatically fit the workflow - we have a 'fit_transform' function that does this if the user wants to do both fit and transform - and have explicitly called out in the docs that we expect the workflow to be have fit called before running transform: 
https://github.com/NVIDIA/NVTabular/blob/e9c0795ab68d4d42b13d3a73e888c73b1483fb73/nvtabular/workflow/workflow.py#L100-L102

My worry about merging the fit & transform functionality is if the user accidentally tries to call transform on a validation dataset, on a workflow that hasn't been fit already. While this is fine for the schema (since there isn't actual data) - this will silently do the wrong thing if we fit the full workflow here.